network/src/sync/blocks.rs

```diff
@@ -107,27 +93,22 @@ impl<S: Storage + Send + std::marker::Sync + 'static> Node<S> {
                 return Err(error.into());
             }
         };
+        let block_struct = <Block<Transaction<Components>> as VMBlock>::serialize(&block_struct)?;
+
+        let canon = self.storage.canon().await?;
 
         info!(
             "Received block from {} of epoch {} with hash {} (current head {})",
             remote_address,
             block_struct.header.time,
-            block_struct.header.get_hash(),
-            self.expect_sync().current_block_height(),
+            block_struct.header.hash(),
+            canon.block_height,
         );
 
         // Verify the block and insert it into the storage.
-        let block_validity = self.expect_sync().consensus.receive_block(&block_struct).await;
+        let block_validity = self.expect_sync().consensus.receive_block(block_struct).await;
 
-        if let Err(ConsensusError::PreExistingBlock) = block_validity {
-            if is_block_new {
-                metrics::increment_counter!(DUPLICATE_BLOCKS);
-            } else {
-                metrics::increment_counter!(DUPLICATE_SYNC_BLOCKS);
```

Comment:
This metric is not re-introduced anywhere, i.e. it gets broken; this piece of code can still remain here instead of putting the `DUPLICATE_BLOCKS` bumps directly under `receive_block`.

Reply:
see consensus/inner.rs:252. doesn't cover sync blocks though -- that was a decision i made to keep it simple. let me know if you disagree

Reply:
It would be practical to know if we are receiving duplicate sync blocks though, as we still do get them; if it does make the related code spaghetti, we should ditch the related metric and keep the issue in mind, but perhaps it has become simpler (to pass this information on) than it was initially? It's pretty useful for diagnostics, especially at this stage of development.